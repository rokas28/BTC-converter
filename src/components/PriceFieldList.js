import React from "react";
import PriceField from "./PriceField";

const PriceFieldList = ({ value, currencies, priceFields, removeField }) => {
  const renderedPriceFields = currencies.map((currency) => {
    if (!priceFields.includes(currency[0])) {
      return null;
    }
    return (
      <PriceField
        key={currency[0]}
        value={value}
        currency={currency[1]}
        removeField={removeField}
      />
    );
  });

  return <div>{renderedPriceFields}</div>;
};

export default PriceFieldList;
