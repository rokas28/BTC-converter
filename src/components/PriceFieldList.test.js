import React from "react";
import renderer from "react-test-renderer";
import PriceFieldList from "./PriceFieldList";

const currency = [
  [
    "USD",
    {
      code: "USD",
      symbol: "&#36;",
      rate: "47,520.6658",
      description: "United States Dollar",
      rate_float: 47315.0247,
    },
  ],
  [
    "GBP",
    {
      code: "GBP",
      symbol: "&pound;",
      rate: "34,023.3156",
      description: "British Pound Sterling",
      rate_float: 33926.6234,
    },
  ],
  [
    "EUR",
    {
      code: "EUR",
      symbol: "&euro;",
      rate: "39,370.2564",
      description: "Euro",
      rate_float: 39303.5501,
    },
  ],
];

it("shoud render PriceFieldList with USD field with right currency", () => {
  const value = 2;
  const priceFields = ["USD"];

  const tree = renderer
    .create(
      <PriceFieldList
        value={value}
        currencies={currency}
        priceFields={priceFields}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("shoud render PriceFieldList with GBP field with right currency", () => {
  const value = 2;
  const priceFields = ["GBP"];

  const tree = renderer
    .create(
      <PriceFieldList
        value={value}
        currencies={currency}
        priceFields={priceFields}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("shoud render PriceFieldList with  EUR field with right currency", () => {
  const value = 2;
  const priceFields = ["EUR"];

  const tree = renderer
    .create(
      <PriceFieldList
        value={value}
        currencies={currency}
        priceFields={priceFields}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("shoud render PriceFieldList with USD and GBP fields with right currencies", () => {
  const value = 2;
  const priceFields = ["USD", "GBP"];

  const tree = renderer
    .create(
      <PriceFieldList
        value={value}
        currencies={currency}
        priceFields={priceFields}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("shoud render PriceFieldList with USD and EUR fields with right currencies", () => {
  const value = 2;
  const priceFields = ["USD", "EUR"];

  const tree = renderer
    .create(
      <PriceFieldList
        value={value}
        currencies={currency}
        priceFields={priceFields}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("shoud render PriceFieldList with GBP and EUR fields with right currencies", () => {
  const value = 2;
  const priceFields = ["GBP", "EUR"];

  const tree = renderer
    .create(
      <PriceFieldList
        value={value}
        currencies={currency}
        priceFields={priceFields}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("shoud render PriceFieldList with USD, GBP and EUR fields with right currencies", () => {
  const value = 2;
  const priceFields = ["USD", "GBP", "EUR"];

  const tree = renderer
    .create(
      <PriceFieldList
        value={value}
        currencies={currency}
        priceFields={priceFields}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
