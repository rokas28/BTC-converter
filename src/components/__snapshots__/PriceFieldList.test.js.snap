// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud render PriceFieldList with  EUR field with right currency 1`] = `
<div>
  <div
    className="PriceField"
  >
    <p>
      €
      78,607.1
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
</div>
`;

exports[`shoud render PriceFieldList with GBP and EUR fields with right currencies 1`] = `
<div>
  <div
    className="PriceField"
  >
    <p>
      £
      67,853.25
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="PriceField"
  >
    <p>
      €
      78,607.1
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
</div>
`;

exports[`shoud render PriceFieldList with GBP field with right currency 1`] = `
<div>
  <div
    className="PriceField"
  >
    <p>
      £
      67,853.25
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
</div>
`;

exports[`shoud render PriceFieldList with USD and EUR fields with right currencies 1`] = `
<div>
  <div
    className="PriceField"
  >
    <p>
      $
      94,630.05
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="PriceField"
  >
    <p>
      €
      78,607.1
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
</div>
`;

exports[`shoud render PriceFieldList with USD and GBP fields with right currencies 1`] = `
<div>
  <div
    className="PriceField"
  >
    <p>
      $
      94,630.05
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="PriceField"
  >
    <p>
      £
      67,853.25
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
</div>
`;

exports[`shoud render PriceFieldList with USD field with right currency 1`] = `
<div>
  <div
    className="PriceField"
  >
    <p>
      $
      94,630.05
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
</div>
`;

exports[`shoud render PriceFieldList with USD, GBP and EUR fields with right currencies 1`] = `
<div>
  <div
    className="PriceField"
  >
    <p>
      $
      94,630.05
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="PriceField"
  >
    <p>
      £
      67,853.25
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="PriceField"
  >
    <p>
      €
      78,607.1
    </p>
    <svg
      className="close"
      fill="none"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="18"
        x2="6"
        y1="6"
        y2="18"
      />
      <line
        x1="6"
        x2="18"
        y1="6"
        y2="18"
      />
    </svg>
  </div>
</div>
`;
